prov/util/src/util_cntr.c

```diff
@@ -390,7 +390,9 @@ int ofi_cntr_init(const struct fi_provider *prov, struct fid_domain *domain,
 			return ret;
 	}
 
-	ofi_mutex_init(&cntr->ep_list_lock);
+	ofi_genlock_init(&cntr->ep_list_lock,
+			 cntr->domain->threading != FI_THREAD_SAFE ?
+			 OFI_LOCK_NOOP : OFI_LOCK_MUTEX);
```

Comment:
Can only disable for FI_THREAD_DOMAIN or FI_THREAD_COMPLETION

Reply:
done